napari/components/_viewer_mouse_bindings.py

```diff
@@ -0,0 +1,15 @@
+def dims_scroll(viewer, event):
+    """Scroll the dimensions slider."""
+    if 'Control' not in event.modifiers:
+        return
+    if event.native.inverted():
+        viewer.dims._scroll_progress += event.delta[1]
+    else:
+        viewer.dims._scroll_progress -= event.delta[1]
+    if abs(viewer.dims._scroll_progress) >= 1:
+        for i in range(int(abs(viewer.dims._scroll_progress))):
+            if viewer.dims._scroll_progress < 0:
+                viewer.dims._increment_dims_left()
+            else:
+                viewer.dims._increment_dims_right()
+        viewer.dims._scroll_progress = 0
```

Comment:
Rather than resetting to 0, this should be

```suggestion
        viewer.dims._scroll_progress %= 1
```

Reply:
This won't get the `-` case correct, if < 0 need to also subtract `1` after modulus operation

Reply:
yep, I am going to update the loop to do so

Reply:
I updated the calculation loop, hopefully addressed this comment and the other concern about residual value

Reply:
Update to calculation looks good @ziyangczi, but I think i'd still put the reset in the control button release and not in the mouse wheel event. The use case I'm imagining is someone presses control they scroll, an increment is left, they release, then the press control and start scrolling again and that increment is used. The approach of putting the reset in the the mouse wheel event with no control press won't reset the increment in this case, but if we put it on the control release then it will reset. Does that make sense?

Reply:
added in the keybinding section. I am slightly concerned since this method touches Viewer, instead of ViewerModel, but this should reset things cleanly

Reply:
> I am slightly concerned since this method touches Viewer, instead of ViewerModel

Yes you are right to notice that this is slightly weird. We might actually want to pull most of those keybindings out into the ViewerModel and just leave https://github.com/napari/napari/blob/master/napari/_viewer_key_bindings.py#L5 there, which I think is the only one that really needs to be there. We can do that in a new PR though